src/app/doctor/doctor.cpp

```diff
@@ -10,7 +10,7 @@ Doctor::Doctor(const std::string& name, const std::string& surname, const std::s
 
 void Doctor::appendVisit(const std::shared_ptr<Visit>& visit)
 {
-    visits_.push_back(visit);
+    visits_.emplace_back(visit);
 }
```

Comment:
Functions that add associations should also make a backward connection to an other class

Reply:
Doctor does not need that. 
Given Visit contruction doctors is assigned, then only doctor needs to know about the visit.